src/components/Map.js

```diff
@@ -42,11 +54,23 @@ const Map = () => {
                         `${process.env.PUBLIC_URL}/placeholder.png`
                       }
                       alt={location?.thumbnail?.source ? location.title : ''}
+                      onClick={() => {
+                        handleClick(location);
+                      }}
                     />
                   );
                 })
               : null}
           </GoogleMapReact>
+
+          {locationDetails ? (
+            <Dialog className="dialog" aria-label="Location details">
```

Comment:
The Reach modal/dialog has built-in to close when the user presses the escape key or clicks outside of the modal. This one isn't, and I'm wondering if it's because we're not closing the modal with `onDismiss` https://reach.tech/dialog. We could take advantage of that prop and the `show` prop, I'm guessing.

Reply:
Done! Thank you for the suggestion :)